app.py

```diff
@@ -102,357 +125,270 @@ def toggle_left_column(left_column_collapse: int, class_name: str) -> str:
-        return empty_figure, "tab"
-    if ctx.triggered_id == "run-button":
-        if "MIP" in solver:
-            model_data = JobShopData()
-            filename = SCENARIOS[scenario]
-            model_data.load_from_file(DATA_PATH.joinpath(filename), resource_names=RESOURCE_NAMES)
-            use_mip_solver = True
-            allow_quadratic_constraints = model == "QM"
-            if allow_quadratic_constraints:
-                time.sleep(0.1)  # sleep to allow the loading icon to appear first
-                fig = get_empty_figure(HTML_CONFIGS["solver_messages"]["mip"]["quadratic_error"])
-                class_name = "tab-fail"
-                mip_table = generate_output_table(0, 0, 0)
-                mip_table_style = {"visibility": "hidden"}
-            else:
-                start = time.time()
-                results = run_shop_scheduler(
-                    model_data,
-                    use_mip_solver=use_mip_solver,
-                    allow_quadratic_constraints=allow_quadratic_constraints,
-                    solver_time_limit=time_limit,
-                )
-                end = time.time()
-                if len(results) == 0:
-                    fig = get_empty_figure(HTML_CONFIGS["solver_messages"]["mip"]["no_solution"])
-                    mip_table = generate_output_table(0, 0, 0)
-                    class_name = "tab-fail"
-                    mip_table_style = {"visibility": "hidden"}
-                else:
-                    fig = generate_gantt_chart(df=results, y_axis="Job", color="Resource")
-                    class_name = "tab-success"
-                    mip_table = generate_output_table(
-                        results["Finish"].max(), time_limit, int(end - start)
-                    )
-                    mip_table_style = {"visibility": "visible"}
-            return fig, mip_table, class_name, mip_table_style
-        else:
-            time.sleep(0.1)  # sleep to allow the loading icon to appear first
-            message = HTML_CONFIGS["solver_messages"]["mip"]["solver_not_chosen"]
-            empty_figure = get_empty_figure(message)
-            mip_table = generate_output_table(0, 0, 0)
-            mip_table_style = {"visibility": "hidden"}
-            return empty_figure, mip_table, "tab-warning", mip_table_style
-    else:
+    if ctx.triggered_id != "run-button" or run_click == 0:
         raise PreventUpdate
 
+    if not SamplerType.MIP.value in solvers:
+        return (dash.no_update, dash.no_update, "tab", "Classical Results", True, False)
+
+    model_data = JobShopData()
+    filename = SCENARIOS[scenario]
+    model_data.load_from_file(DATA_PATH.joinpath(filename), resource_names=RESOURCE_NAMES)
+
+    start = time.time()
+    results = run_shop_scheduler(
+        model_data,
+        use_mip_solver=True,
+        allow_quadratic_constraints=False,
+        solver_time_limit=time_limit,
+    )
+    end = time.time()
+    if len(results):
```

Comment:
This might make more sense here:
```suggestion
    if results.empy():
```